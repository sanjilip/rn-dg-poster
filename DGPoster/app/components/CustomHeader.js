import { View, Text, Image, TouchableOpacity } from 'react-native';

import { customHeaderStyle as style } from './style';

const CustomHeader = ({
    title,
    leftIcon,
    rightIcon,
    backgroundImage,
    rightActionCalled,
    leftActionCalled
}) => {
    return (
        <View style={style.headerContainer}>
            <Image
                source={backgroundImage}
                style={style.backgroundImage}
                resizeMode="cover"
            />
            <View style={style.innerContainer}>
                <TouchableOpacity onPress={leftActionCalled}>
                    <Image source={leftIcon} style={style.leftIcon} />
                </TouchableOpacity>
                <Text style={style.title}>{title}</Text>
                <TouchableOpacity
                    style={style.rightIcon}
                    onPress={rightActionCalled}>
                    <Image source={rightIcon} style={style.rightIcon} />
                </TouchableOpacity>
            </View>
        </View>
    );
};

export default CustomHeader;
