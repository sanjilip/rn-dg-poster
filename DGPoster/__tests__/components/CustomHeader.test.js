import React from 'react';
import { shallow } from 'enzyme';

import CustomHeader from '../../app/components/CustomHeader';

describe('Test CustomHeader', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<CustomHeader />);
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
