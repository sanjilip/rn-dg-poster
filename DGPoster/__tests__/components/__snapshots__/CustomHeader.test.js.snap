// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CustomHeader renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "black",
      "height": 40,
      "marginBottom": 10,
    }
  }
>
  <Image
    resizeMode="cover"
    style={
      {
        "bottom": 0,
        "left": 0,
        "opacity": 0.5,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "paddingHorizontal": 10,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <TouchableOpacity>
      <Image
        style={
          {
            "height": 18,
            "marginRight": 15,
            "width": 18,
          }
        }
      />
    </TouchableOpacity>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    />
    <TouchableOpacity
      style={
        {
          "alignItems": "flex-end",
          "flex": 1,
          "height": 23,
          "width": 23,
        }
      }
    >
      <Image
        style={
          {
            "alignItems": "flex-end",
            "flex": 1,
            "height": 23,
            "width": 23,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;
